src/main/java/org/spongepowered/common/mixin/command/vanilla/MixinCommandWorldBorder.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of Sponge, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.common.mixin.command.vanilla;
+
+import net.minecraft.command.CommandWorldBorder;
+import net.minecraft.world.WorldServer;
+import net.minecraft.world.border.WorldBorder;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Redirect;
+import org.spongepowered.common.SpongeImpl;
+
+@Mixin(CommandWorldBorder.class)
+public abstract class MixinCommandWorldBorder {
```

Comment:
Its cause the server sends the correct coordinates down (not ones based on Nether's scaled by 8 system) and then the client takes it and divides it by 8 to render the border. We lie to the client so the WorldBorder works correctly client-side.

Now thinking about things, you are correct that I do return a ```new WorldBorder()``` which is meant for the ```WorldProviderHell```. With that known, we don't want to apply my packet hack when SF is installed on the client as the coordinates will now be off by 8 (as WorldProviderHell will return a non-adjusting WorldBorder for the Server and Client World).

Can you address this @Minecrell ? Test it first to see.